newroutes/homepage.js

```diff
@@ -0,0 +1,77 @@
+const autoBind = require("auto-bind")
+const express = require("express")
+
+// Import custom error types
+const { BadRequestError } = require("@errors/BadRequestError")
+
+const {
+  attachReadRouteHandlerWrapper,
+  attachWriteRouteHandlerWrapper,
+} = require("@middleware/routeHandler")
+
+const { UpdateHomepageSchema } = require("@validators/RequestSchema")
+
+const { authMiddleware } = require("@root/newmiddleware/index")
+
+class HomepageRouter {
+  constructor({ homepagePageService }) {
+    this.homepagePageService = homepagePageService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  // Read homepage index file
+  async readHomepage(req, res) {
+    const { accessToken } = req
+
+    const { siteName } = req.params
+
+    const readResp = await this.homepagePageService.read({
+      siteName,
+      accessToken,
+    })
+
+    return res.status(200).json(readResp)
+  }
+
+  // Update homepage index file
+  async updateHomepage(req, res) {
+    const { accessToken } = req
+
+    const { siteName } = req.params
+    const { error } = UpdateHomepageSchema.validate(req.body, {
+      allowUnknown: true,
+    })
+    if (error) throw new BadRequestError(error.message)
+    const {
+      content: { frontMatter, pageBody },
+      sha,
+    } = req.body
+
+    const updateResp = await this.homepagePageService.update(
```

Comment:
could we maybe rename this to `updatedHomepage`? not sure `updateResp` is entirely clear

Reply:
I've updated the variable name, https://github.com/isomerpages/isomercms-backend/pull/331/commits/65f497e0aa5706b6a0648d0e9a2de6e3a2ac893e